tests/benchmark/README.md

```diff
@@ -0,0 +1,320 @@
+
+### Definition:  
+`BENCHMARK_TEMPLATE_DEFINE_F(fixture_name, test_name, data type)(benchmark::State &st){fixture_name::test_implementation(st, test_args);}`  
+In the definition we:
+1. Bind the test to a specific fixture
+2. Give it a unique name (see [Tests naming](#tests-naming))
+3. Specify the fixture data type (currently available: `fp32_index_t` and `fp64_index_t`) (see [Fixture template type](#fixture-template-type).  
+4. Call the function, defined in the fixture class, that implements the required test case.
+### Registration
+`BENCHMARK_REGISTER_F(test_fixture, test_name)->Args({val1, val2})->ArgNames({"name1", "name2"})->Iterations(val0)->Unit(benchmark::kMillisecond)`  
+Here we register to the benchmarks running loop the test that was bind to `test_fixture` and has `test_name` .  
+*NOTE:* The **registration order** is important. Changing it might violate the ability to compare between one benchmark run to another. The reason to that is that we have tests (such as addLabel and DeleteLabel) that modify the hnsw graph.
+# Fixtures
+## Fixture template type
+Since Google benchmarks supports fixture with only one template argument, we use a helper struct to define multiple types.  
+```asm
+// benchmarks/bm_definitions.h
+template <VecSimType type, typename DataType, typename DistType = DataType>
+struct IndexType {
+
+    static VecSimType get_index_type() { return type; }
+
+    typedef DataType data_t;
+    typedef DistType dist_t;
+};
+
+// benchmarks/bm_vecsim_index.h
+template <typename index_type_t>
+class BM_VecSimIndex : public BM_VecSimGeneral {
+    
+    // Parsing the struct's types
+    using data_t = typename index_type_t::data_t;
+    using dist_t = typename index_type_t::dist_t;
+    ...
+    static std::vector<std::vector<data_t>> queries;
+
+    // Using the VecSimType getter 
+    static void Initialize() {
+        VecSimType type = index_type_t::get_index_type();
+        ...
+    }
+};
+
+```
+## Benchmarks fixture classes hierarchy 
+You can find the hierarchy graph, defining the main members of each class here:  
+[Benchmarks directory hierarchy graph](https://docs.google.com/presentation/d/1_vqZ50--H9Jv1yf-Pczq30n-kjtO9LW_XYOuH-D0Xjk/edit#slide=id.p).
+
+### class BM_VecSimGeneral
+Defined in bm_vecsim_general.h  
+This class inherits from includes all the benchmark::Fixture.  
+It includes the index parameters as static data members. The static data members are initialized in the .cpp files under benchmark/run_files.  
+To refer a member from one of the derived classes, BM_VecSimGeneral:: must be used. Shortcut macros such as `#define N_QUERIES BM_VecSimGeneral::n_queries` are availble in bm_definitions.h.
+### Remarks
+* block_size is public because it is used to define the number of iterations on some test cases
+
+### class BM_VecSim_Index<index_type_t> : public BM_VecSimGeneral
```

Comment:
please remove inheritance from the titles, refer to the class and the illustration in the text